### PR TITLE
Add note to withdraw

### PR DESCRIPTION
**Pull request by mkalenska**

### CODE REVIEW

File: app/api/v2/account/withdraws.rb [hunk 1]
```diff
@@ -53,6 +53,9 @@ class Withdraws < Grape::API
                    type: { value: BigDecimal, message: 'account.withdraw.non_decimal_amount' },
                    values: { value: ->(v) { v.try(:positive?) }, message: 'account.withdraw.non_positive_amount' },
                    desc: 'The amount to withdraw.'
+          optional :note,
```
Comment (ysv) [inline app/api/v2/account/withdraws.rb:56]: add validation to note length

File: app/api/v2/account/withdraws.rb [hunk 2]
```diff
@@ -66,7 +69,8 @@ class Withdraws < Grape::API
             sum:            params[:amount],
             member:         current_user,
             currency:       currency,
-            rid:            params[:rid]
+            rid:            params[:rid],
+            note:           params[:note]
           withdraw.save!
           withdraw.with_lock { withdraw.submit! }
           present withdraw, with: API::V2::Entities::Withdraw
```
Comment (ysv) [inline app/api/v2/account/withdraws.rb:77]: add note to `API::V2::Entities::Withdraw`

File: app/models/withdraws/coin.rb [hunk 3]
```diff
@@ -65,6 +65,7 @@ def as_json_for_event_api
 #  type         :string(30)       not null
 #  tid          :string(64)       not null
 #  rid          :string(95)       not null
+#  note         :string(20)
```
Comment (ysv) [inline app/models/withdraws/coin.rb]: Why is it 20 chars max ?

File: spec/api/v2/account/withdraws_spec.rb [hunk 4]
```diff
@@ -212,6 +212,18 @@
         expect(record.account.balance).to eq(1.2 - amount)
         expect(record.account.locked).to eq amount
       end
+
+      it 'creates new withdraw with note' do
```
Comment (ysv) [inline spec/api/v2/account/withdraws_spec.rb:217]: add spec for response body

**Review by ysv — CHANGES_REQUESTED** [review]

File: app/api/v2/account/withdraws.rb [hunk 5]
```diff
@@ -53,6 +53,10 @@ class Withdraws < Grape::API
                    type: { value: BigDecimal, message: 'account.withdraw.non_decimal_amount' },
                    values: { value: ->(v) { v.try(:positive?) }, message: 'account.withdraw.non_positive_amount' },
                    desc: 'The amount to withdraw.'
+          optional :note,
+                   type: String,
+                   values: { value: ->(v) { v.size <= 20 }, message: 'account.withdraw.too_long_note' },
```
Comment (ysv) [inline app/api/v2/account/withdraws.rb]: size can be 20
```
v.size < 20
```

File: app/models/withdraw.rb [hunk 6]
```diff
@@ -274,6 +275,7 @@ def send_coins!
 #  type         :string(30)       not null
 #  tid          :string(64)       not null
 #  rid          :string(95)       not null
+#  note         :string(20)
```
Comment (ysv) [inline app/models/withdraw.rb]: put maximum length 256

Comment (ysv) [inline app/models/withdraw.rb]: @maryna25 put 256 max note size same as [Bitgo](https://www.bitgo.com/api/v2/#operation/v2.wallet.tx.build) does

**Review by ysv — CHANGES_REQUESTED** [review]

File: app/models/withdraw.rb [hunk 7]
```diff
@@ -33,6 +33,7 @@ class Withdraw < ApplicationRecord
   validates :sum,
             presence: true,
             numericality: { greater_than_or_equal_to: ->(withdraw) { withdraw.currency.min_withdraw_amount }}
+  validates :note, length: { maximum: 20 }, allow_blank: true
```
Comment (ysv) [inline app/models/withdraw.rb]: I think it's validated automatically by `validates_lengths_from_database` gem
But check that it's true

**Review by ysv — CHANGES_REQUESTED** [review]

File: app/models/withdraw.rb [hunk 8]
```diff
@@ -274,6 +274,7 @@ def send_coins!
 #  type         :string(30)       not null
 #  tid          :string(64)       not null
 #  rid          :string(95)       not null
+#  note         :string(256)
```
Comment (ysv) [inline app/models/withdraw.rb:277]: Single thing is missing is spec for long note in models

File: app/api/v2/account/withdraws.rb [hunk 9]
```diff
@@ -53,6 +53,10 @@ class Withdraws < Grape::API
                    type: { value: BigDecimal, message: 'account.withdraw.non_decimal_amount' },
                    values: { value: ->(v) { v.try(:positive?) }, message: 'account.withdraw.non_positive_amount' },
                    desc: 'The amount to withdraw.'
+          optional :note,
+                   type: String,
+                   values: { value: ->(v) { v.size <= 256 }, message: 'account.withdraw.too_long_note' },
+                   desc: 'Withdraw note.'
```
Comment (ysv) [inline app/api/v2/account/withdraws.rb]: put something like this in desc
'Optional metadata to be applied to the transaction. Typically used to 'tag' transactions with memorable comments.'

**Review by ysv — APPROVED** [review]

**Review by mod — APPROVED** [review]

**Pull request merged**